src/bootstrap/test.rs

```diff
@@ -1636,8 +1636,9 @@ impl Step for Crate {
         // libstd, then what we're actually testing is the libstd produced in
         // stage1. Reflect that here by updating the compiler that we're working
         // with automatically.
+        // FIXME(mati865): do similar for the other branch if this is correct
         let compiler = if builder.force_use_stage1(compiler, target) {
-            builder.compiler(1, compiler.host)
+            builder.compiler(1, builder.config.build)
```

Comment:
It solved the error I mentioned in OP where `proc-macro` dependencies were located in `obj/build/x86_64-unknown-linux-gnu/stage1-test/x86_64-unknown-linux-musl/release/deps` but rustc was looking inside `obj/build/x86_64-unknown-linux-musl/stage1-test/x86_64-unknown-linux-musl/release/deps`.
I could be wrong but probably it's the first time we want to run tests built for the different host.

Reply:
This one is wrong because it makes tests run from `gnu` host. That makes `bootstrap` build stage1 `gnu` toolchain and use it for the test with `musl` target.
We don't want to waste time on building stage1 `gnu` host.